clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -1287,6 +1287,16 @@ public void testCommitAsyncNegativeOffset() {
         assertTrue(mockOffsetCommitCallback.exception instanceof IllegalArgumentException);
     }
 
+    @Test(expected = CommitFailedException.class)
+    public void testCommitOffsetSyncCallbackWithoutNonRetriableException() {
```

Comment:
Thanks for the review @hachikuji , in the test we need to make sure that future gets failed (due to any possible error, in this case it is UNKNOWN_MEMBER_ID) and offset commit sync request throws the appropriate exception when it discovers that the future is failed and it is non-retriable also. That is why we added below to prevent the message mentioned in the JIRA : 

`            if (future.failed() && !future.isRetriable())
`

I see that currently we don't try to commit the offsets if timeout occurs. Please let me know if I am missing something. 

@ijuma please correct me if I am wrong.